specutils/tests/test_loaders.py

```diff
@@ -1273,3 +1273,85 @@ def test_galah_guess(remote_data_path):
         assert isinstance(spectra[1].uncertainty, StdDevUncertainty)
         assert spectra[1].meta.get("label") is not None
         assert spectra[1].meta.get("header") is not None
+
+
+# We cannot use remote_access directly in the MIRI MRS tests, because
+# the test functions are called once for every file in the remote access
+# list, but we need to feed the entire set to the functions at once. We
+# store the file names in a list and use a dummy test method to load the
+# list via the remote_access machinery.
+#
+# MIRI MRS 1D data sets are comprised of 12 files.
+# We add one bad file to test the skip/warn on missing file functionality
+filename_list = ["bad_file.fits"]
+@remote_access([
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch1-long_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch1-medium_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch1-short_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch2-long_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch2-medium_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch2-short_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch3-long_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch3-medium_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch3-short_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch4-long_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch4-medium_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch4-short_x1d.fits'},
+])
+def test_loaddata_miri_mrs(remote_data_path):
```

Comment:
What is this test supposed to do? I thought this was a fixture but looks like it is its own test function now.

If this is meant to set up the global var for other tests, why not use `setup_module`?

Reply:
@ibusko wrote these tests (I just added to them), perhaps he can shed light on why this was necessary. The answer might be that he wasn't familiar (I'm not either) with `setup_module`, in which case we can change it...

Reply:
Reading back further in the history, I imagine it's this:

> Indeed. But the `@remote_access` decorator doesn't seem to work with setup functions. Apparently it can be applied to `test_` functions only.

Reply:
Can we guarantee that this test function will always run first? What if you enable parallel mode using `pytest-xdist`? 💭

Reply:
After some offline discussions, @rosteen has convinced me that the tests are running as expected.